src/pages/admin/import-video.tsx

```diff
@@ -409,6 +451,16 @@ class IndexApp extends React.Component<Props, State> {
             </div>
         </Modal>
     }
+    renderDeleteVideo() {
+        return <Modal isOpen={this.state.showDeleteVideo}>
+            <div>
+                <div>Enter ID: <input value={this.state.toDeleteVideo} onChange={(item: any) => this.setState({ toDeleteVideo: item.target.value })} /></div>
+                <div>Type &quot;Delete forever&quot;: <input value={this.state.deleteConfirmation} onChange={(item: any) => this.setState({ deleteConfirmation: item.target.value })} /></div>
+                <button style={{background: 'orange'}} onClick={() => { if (this.delete()) this.setState({ showDeleteVideo: false, deleteConfirmation: '', toDeleteVideo: '' }) }}>DELETE</button>
```

Comment:
Can you do the state modification in `this.delete`? For `toDeleteVideo` and `deleteConfirmation` you've already cleared them. Also the return value of delete will always been truthy because it's an unresolved `Promise`.

Reply:
Good points! The state modification already exists in `this.delete` (not sure why I doubled up the code)... Will remove from the `onClick` event.